### DISCUSSION TITLE
安装alist插件并配置，在文章中和附件库无法预览图片

### DISCUSSION THREAD

**Discussion by MrArlong** [halo-dev/halo#7131] [Q&A]
### 系统信息

- 外部访问地址: https://halo..top
- 启动时间: 2024-10-17 14:34
- 版本: 2.20.3
- 构建时间: 2024-10-15 17:26
- Git Commit: b95a83a
- Java: IBM Semeru Runtime Open Edition / 21.0.4+7-LTS
- 数据库: MySQL Community Server - GPL / 8.2.0
- 操作系统: Linux / 5.15.60-1.el7.x86_64
- 已激活主题: [Dream 1.3.2](https://github.com/nineya/halo-theme-dream2.0.git)
- 已启动插件:
  - [AList 存储库 1.1.0](https://github.com/halo-sigs/plugin-alist)
  - [应用市场 1.6.0](https://www.halo.run/store/apps/app-VYJbF)
  - [瞬间 1.8.0](https://github.com/halo-sigs/plugin-moments)
  - [ByteMD 1.1.1](https://github.com/ruibaby/plugin-bytemd)
  - [图库管理 1.4.0](https://github.com/halo-sigs/plugin-photos)
  - [搜索组件 1.6.2](https://github.com/halo-dev/plugin-search-widget)
  - [RSS 1.1.1](https://github.com/halo-sigs/plugin-feed)
  - [评论组件 2.5.0](https://github.com/halo-dev/plugin-comment-widget)
  - [Sitemap 1.0.2](https://github.com/halo-sigs/plugin-sitemap)

### 使用的哪种方式运行？

Docker

### 发生了什么？

我配置了alist之后发现上传的图片无法回显
![Image](https://github.com/user-attachments/assets/dcfc8bea-c5d9-437d-9277-bfd420e0887f)

### 复现步骤

点开附件，点击上传，选择alist，上传，alist正常显示文件，但是从halo里面点预览发现预览失败

**Comment by JohnNiang** [comment]
Hi @MrArlong , thank you for reaching out here!

1. 直接访问这个链接会发生什么错误？
2. 需要提供完整的复现步骤。

**Comment by MrArlong** [comment]
@JohnNiang 
1.访问这个链接会直接下载我上传的附件
2.
第一步：安装alist存储库插件    [AList 存储库 1.1.0](https://github.com/halo-sigs/plugin-alist)
第二步：在附件库新建存储策略，并填写alist信息，点击AList 存储验证，验证成功
第三步：在附件库点击右上角上传按钮，选择刚刚新建的alist存储库，点击上传
第四步：点击上传的附件，弹窗显示预览异常，见图
![Image](https://github.com/user-attachments/assets/cda09df2-d432-458d-8f9a-06660793b308)


第五步：访问链接会直接下载刚刚的附件
第六步：去alist网站可以正常查看该图片，与halo链接不同的是alist链接把特殊符号进行了转译
![Image](https://github.com/user-attachments/assets/6f852f4e-e9af-4cea-9fa1-7785a498bbe6)

![Image](https://github.com/user-attachments/assets/7f314a71-0c72-471b-b2ad-4b93241fa288)

**Comment by JohnNiang** [comment]
Hi @MrArlong ，感谢进一步反馈。

如果确认是 URL 编码的问题。我这边尝试复现一下。

**Comment by JohnNiang** [comment]
Hi @MrArlong ，建议提供以下完整的链接。我这边确认一下是不是阿里云盘“防盗链”功能导致的。或者你可以根据[这个评论](https://github.com/halo-sigs/plugin-alist/issues/29#issuecomment-2441381663)自行测试。

**Comment by MrArlong** [comment]
@JohnNiang 你好，这是我上传halo的图片链接，我也看不出来他们两个到底有什么区别
`https://cn-beijing-data.aliyundrive.net/H3cExCdv%2F2294138920%2F6715f9e65528b5371df44b318f87ff6594526434%2F6715f9e63faef41b7fda433589360ceb1253fc2c?callback=eyJjYWxsYmFja1VybCI6Imh0dHA6Ly9iajI5LmFwaS1ocC5hbGl5dW5wZHMuY29tL3YyL2ZpbGUvZG93bmxvYWRfY2FsbGJhY2siLCJjYWxsYmFja0JvZHkiOiJodHRwSGVhZGVyLnJhbmdlPSR7aHR0cEhlYWRlci5yYW5nZX1cdTAwMjZidWNrZXQ9JHtidWNrZXR9XHUwMDI2b2JqZWN0PSR7b2JqZWN0fVx1MDAyNmRvbWFpbl9pZD0ke3g6ZG9tYWluX2lkfVx1MDAyNnVzZXJfaWQ9JHt4OnVzZXJfaWR9XHUwMDI2ZHJpdmVfaWQ9JHt4OmRyaXZlX2lkfVx1MDAyNmZpbGVfaWQ9JHt4OmZpbGVfaWR9XHUwMDI2cGRzX3BhcmFtcz0ke3g6cGRzX3BhcmFtc31cdTAwMjZ2ZXJzaW9uPSR7eDp2ZXJzaW9ufSIsImNhbGxiYWNrQm9keVR5cGUiOiJhcHBsaWNhdGlvbi94LXd3dy1mb3JtLXVybGVuY29kZWQiLCJjYWxsYmFja1N0YWdlIjoiYmVmb3JlLWV4ZWN1dGUiLCJjYWxsYmFja0ZhaWx1cmVBY3Rpb24iOiJpZ25vcmUifQ%3D%3D&callback-var=eyJ4OmRvbWFpbl9pZCI6ImJqMjkiLCJ4OnVzZXJfaWQiOiI3MTgzNmU3MGFhZjM0OTExOWI0NDRmMmNkN2U1OTZjMiIsIng6ZHJpdmVfaWQiOiIyMjk0MTM4OTIwIiwieDpmaWxlX2lkIjoiNjcyMDdlYjFkMmM5ZDgxMDI1ZDA0ZGE1ODJiMzg5ZDEyNGQwNjExMiIsIng6cGRzX3BhcmFtcyI6IntcImFwXCI6XCI3NjkxN2NjY2NkNDQ0MWMzOTQ1N2EwNGY2MDg0ZmIyZlwifSIsIng6dmVyc2lvbiI6InYzIn0%3D&di=bj29&dr=2294138920&f=67207eb1d2c9d81025d04da582b389d124d06112&pds-params=%7B%22ap%22%3A%2276917ccccd4441c39457a04f6084fb2f%22%7D&response-content-disposition=attachment%3B%20filename%2A%3DUTF-8%27%271%2520%25281%2529.jpg&security-token=CAISvgJ1q6Ft5B2yfSjIr5fmKsjmqa9KgZuoUB%2FShmQ%2Beth%2FnLDIiTz2IHhMf3NpBOkZvvQ1lGlU6%2Fcalq5rR4QAXlDfNRTNJhOSq1HPWZHInuDox55m4cTXNAr%2BIhr%2F29CoEIedZdjBe%2FCrRknZnytou9XTfimjWFrXWv%2Fgy%2BQQDLItUxK%2FcCBNCfpPOwJms7V6D3bKMuu3OROY6Qi5TmgQ41Uh1jgjtPzkkpfFtkGF1GeXkLFF%2B97DRbG%2FdNRpMZtFVNO44fd7bKKp0lQLs0ARrv4r1fMUqW2X543AUgFLhy2KKMPY99xpFgh9a7j0iCbSGyUu%2FhcRm5sw9%2Byfo34lVYne4zLH%2By597uHwufJ7FxfIREfquk63pvSlHLcLPe0Kjzzleo2k1XRPVFF%2B535IaHXuToXDnvSip35io%2FXtuMkagAFKN7mmrjSHnNYJQutwnD93IKmTr0NZepuyULMYydT9x%2BF7muzzlDPxGCO5MCpATxkxJHozTMUfxH%2BgLZfXTrWEAwUoeHrJibQ5yUYbdHIHNfO3mAJZGJmsiLtuGQXeD6jaPCGITapWptQCXM874C9eugXq9Y%2B1tN0fOHSa0ahltCAA&u=71836e70aaf349119b444f2cd7e596c2&x-oss-access-key-id=STS.NTSarRDpk6YCR9cbdkvWSsuck&x-oss-expires=1730197235&x-oss-signature=PPPeJ2%2ByudVKB5AJ%2FjKOmx4%2BbKmU4CgL8%2BsyLIROmaM%3D&x-oss-signature-version=OSS2`



下面是alist里面的链接

`https://cn-beijing-data.aliyundrive.net/H3cExCdv%2F2294138920%2F6715f9e65528b5371df44b318f87ff6594526434%2F6715f9e63faef41b7fda433589360ceb1253fc2c?callback=eyJjYWxsYmFja1VybCI6Imh0dHA6Ly9iajI5LmFwaS1ocC5hbGl5dW5wZHMuY29tL3YyL2ZpbGUvZG93bmxvYWRfY2FsbGJhY2siLCJjYWxsYmFja0JvZHkiOiJodHRwSGVhZGVyLnJhbmdlPSR7aHR0cEhlYWRlci5yYW5nZX1cdTAwMjZidWNrZXQ9JHtidWNrZXR9XHUwMDI2b2JqZWN0PSR7b2JqZWN0fVx1MDAyNmRvbWFpbl9pZD0ke3g6ZG9tYWluX2lkfVx1MDAyNnVzZXJfaWQ9JHt4OnVzZXJfaWR9XHUwMDI2ZHJpdmVfaWQ9JHt4OmRyaXZlX2lkfVx1MDAyNmZpbGVfaWQ9JHt4OmZpbGVfaWR9XHUwMDI2cGRzX3BhcmFtcz0ke3g6cGRzX3BhcmFtc31cdTAwMjZ2ZXJzaW9uPSR7eDp2ZXJzaW9ufSIsImNhbGxiYWNrQm9keVR5cGUiOiJhcHBsaWNhdGlvbi94LXd3dy1mb3JtLXVybGVuY29kZWQiLCJjYWxsYmFja1N0YWdlIjoiYmVmb3JlLWV4ZWN1dGUiLCJjYWxsYmFja0ZhaWx1cmVBY3Rpb24iOiJpZ25vcmUifQ%3D%3D&callback-var=eyJ4OmRvbWFpbl9pZCI6ImJqMjkiLCJ4OnVzZXJfaWQiOiI3MTgzNmU3MGFhZjM0OTExOWI0NDRmMmNkN2U1OTZjMiIsIng6ZHJpdmVfaWQiOiIyMjk0MTM4OTIwIiwieDpmaWxlX2lkIjoiNjcyMDdlYjFkMmM5ZDgxMDI1ZDA0ZGE1ODJiMzg5ZDEyNGQwNjExMiIsIng6cGRzX3BhcmFtcyI6IntcImFwXCI6XCI3NjkxN2NjY2NkNDQ0MWMzOTQ1N2EwNGY2MDg0ZmIyZlwifSIsIng6dmVyc2lvbiI6InYzIn0%3D&di=bj29&dr=2294138920&f=67207eb1d2c9d81025d04da582b389d124d06112&pds-params=%7B%22ap%22%3A%2276917ccccd4441c39457a04f6084fb2f%22%7D&response-content-disposition=attachment%3B%20filename%2A%3DUTF-8%27%271%2520%25281%2529.jpg&security-token=CAISvgJ1q6Ft5B2yfSjIr5fdCPjeh%2BZrgLCfUkj1oUc4eLtYmLbYjTz2IHhMf3NpBOkZvvQ1lGlU6%2Fcalq5rR4QAXlDfNTiJTxOSq1HPWZHInuDox55m4cTXNAr%2BIhr%2F29CoEIedZdjBe%2FCrRknZnytou9XTfimjWFrXWv%2Fgy%2BQQDLItUxK%2FcCBNCfpPOwJms7V6D3bKMuu3OROY6Qi5TmgQ41Uh1jgjtPzkkpfFtkGF1GeXkLFF%2B97DRbG%2FdNRpMZtFVNO44fd7bKKp0lQLs0ARrv4r1fMUqW2X543AUgFLhy2KKMPY99xpFgh9a7j0iCbSGyUu%2FhcRm5sw9%2Byfo34lVYneo7FA%2FS597uHwufJ7FxfIREfquk63pvSlHLcLPe0Kjzzleo2k1XRPVFF%2B535IaHXuToXDnvSiKzknWPXtuMkagAEmzZ0LTxfTd2U77KT4%2BRtFkQGdSh0NtFxdwh6qT1ryILikKSeIFStnRi316a6mLeW0w4TBwr4zxGmRA0cgb2x6trMCIUS%2FJC79965g2GIeVOdkeKeNHJBTexzdvj6IxE1jzDEfaLjbWNZwpTiY0EZHxxPjKKDkVk68IzW%2BgxBgFiAA&u=71836e70aaf349119b444f2cd7e596c2&x-oss-access-key-id=STS.NThCBjj9J7rtPnDEGmt4twsso&x-oss-expires=1730199382&x-oss-signature=aYFePrYzwxl7U7kav8EYKg%2FY4KIkhAmgmJzIi5ArvvI%3D&x-oss-signature-version=OSS2`

**Comment by MrArlong** [comment]
我不知道是否与这个字段有关下图是halo的请求头，含有referer字段
![Image](https://github.com/user-attachments/assets/94a92926-455a-4be8-943f-2643861764aa)
下图是alist的请求头不含有referer字段
![Image](https://github.com/user-attachments/assets/ab2d94f6-0f4a-42e0-9317-422b2b6c0fc2)

阿里云文档描述这个字段的使用方式
https://help.aliyun.com/zh/oss/user-guide/hotlink-protection/
![Image](https://github.com/user-attachments/assets/181f3025-59f1-4e20-9980-842a61fa36cc)

**Comment by JohnNiang** [comment]
Hi @MrArlong ，这是阿里云盘开启了防盗链策略，目的就是禁止当前这种使用。

AList 之所以可以正常访问图片，我估计是给 `img` 标签添加了 `referrerpolicy=no-referrer` 属性。

> https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/referrerPolicy#no-referrer
